solverdummy/solverdummy.py

```diff
@@ -31,24 +38,41 @@
 
 dimensions = interface.get_dimensions()
 vertices = np.zeros((n, dimensions))
+readData = np.zeros((n, dimensions))
+writeData = np.zeros((n, dimensions))
+
+for x in range(0,n):
+    for y in range(0,dimensions):
+        vertices[x,y] = x
+        readData[x,y] = x 
+        writeData[x,y] = x
 
 vertex_ids = interface.set_mesh_vertices(mesh_id, vertices)
 
+DataRead_ID = interface.get_data_id(DataRead_Name,mesh_id)
+DataWrite_ID = interface.get_data_id(DataWrite_Name,mesh_id)
+
 dt = interface.initialize()
     
 while interface.is_coupling_ongoing():
    
     if interface.is_action_required(precice.action_write_iteration_checkpoint()):
+        interface.write_block_vector_data(DataWrite_ID, data_indices,writeData)  
```

Comment:
Does the write data command need to be executed only when preCICE indicates that a checkpoint is needed to be written?  This works but it could confuse a user as it is different than the [Impicit Coupling Section](https://github.com/precice/precice/wiki/Adapter-Example#implicit-coupling) of the Adapter Example

Reply:
Yes, this looks wrong.

Reply:
We have the same pattern in precice/precice#750.

Reply:
Unless there is a specific reason for doing this in all the solver dummies I think we can change this to replicate the Adapter Example exactly